File: examples/java-player/app/src/main/java/com/mlytics/mlysdk/demo/MainActivity.java
package com.mlytics.mlysdk.demo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import android.os.Bundle;
import android.view.View;

import com.google.android.exoplayer2.ExoPlayer;
import com.google.android.exoplayer2.MediaItem;
import com.google.android.exoplayer2.ui.StyledPlayerView;
import com.mlytics.exoplayer.MLYExoPlayer;
import com.mlytics.mlysdk.driver.MLYDriver;
import com.mlytics.mlysdk.driver.MLYDriverOptions;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

public class MainActivity extends AppCompatActivity {
    private StyledPlayerView playerView = null;
    private AppCompatButton playButton = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        playerView = findViewById(R.id.player_view);

        MLYDriver.INSTANCE.initialize(this.getApplicationContext(), new Function1<MLYDriverOptions, Unit>() {
            @Override
            public Unit invoke(MLYDriverOptions options) {
                options.getClient().setId("cegh8d9j11u91ba1u600");
                return null;
            }
        });

        ExoPlayer player = MLYExoPlayer.Companion.build(playerView);

        playerView.setPlayer(player);

        player.setMediaItem(MediaItem.fromUri("https://vsp-stream.s3.ap-northeast-1.amazonaws.com/HLS/raw/SpaceX.m3u8"));

        playButton = findViewById(R.id.playButton);
        playButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                player.setPlayWhenReady(true);
                player.prepare();
                player.play();
            }
        });
    }
}

File: examples/kotlin-player/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.widget.AppCompatButton
import com.google.android.exoplayer2.C
import com.google.android.exoplayer2.ExoPlayer
import com.google.android.exoplayer2.MediaItem
import com.google.android.exoplayer2.ui.StyledPlayerView
import com.mlytics.exoplayer.MLYExoPlayer
import com.mlytics.mlysdk.driver.MLYDriver

class MainActivity : AppCompatActivity() {
    private val clientId = "cegh8d9j11u91ba1u600"
    private val videoUrlString = "https://vsp-stream.s3.ap-northeast-1.amazonaws.com/HLS/raw/SpaceX.m3u8"
    private val subtitleUrlString = "https://static.gordon.sh/p2sp-test/subtitle_zhtw.vtt"
    private var player: ExoPlayer? = null
    private var playerView: StyledPlayerView? = null
    private var playButton: AppCompatButton? = null

    override fun onDestroy() {
        super.onDestroy()
        MLYDriver.deactivate()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        playerView = findViewById<StyledPlayerView>(R.id.player_view)

        MLYDriver.initialize(this.applicationContext) { options ->
            options.client.id = clientId
        }

        MLYExoPlayer.latency = 13000
        this.player = MLYExoPlayer.build(playerView!!)
        player?.playWhenReady = true

        playerView?.player = player
        playerView?.controllerShowTimeoutMs = -1

        this.playVideo()

        playButton = findViewById(R.id.playButton)
        playButton?.setOnClickListener {
            playVideo()
        }
    }

    private fun playVideo() {
        val videoUri = Uri.parse(videoUrlString)

        val subtitleUri = Uri.parse(subtitleUrlString)
        val subtitleConfiguration = MediaItem.SubtitleConfiguration.Builder(subtitleUri)
            .setMimeType("text/vtt")
            .setLanguage("zh-TW")
            .setSelectionFlags(C.SELECTION_FLAG_DEFAULT)
            .build()

        val mediaItem = MediaItem.Builder()
            .setUri(videoUri)
            .setSubtitleConfigurations(listOf(subtitleConfiguration))
            .build()

        player?.setMediaItem(mediaItem)
        player?.prepare()
        player?.play()
    }
}
